src/pages/Product/Product.js

```diff
@@ -1,19 +1,35 @@
 import "./Product.css";
 import { useEffect } from "react";
 import { Filter, ProductListing } from "../../components";
+import { encodedToken } from "../../utilities/token";
 import { useDocumentTitle } from "../../hooks/useDocumentTitle";
-import { useProducts, useProductsFilter } from "../../contexts";
+import { useCart, useProducts, useProductsFilter } from "../../contexts";
 import {
   getFilteredCategoryProducts,
   getFilteredPriceProducts,
   getFIlteredRatingProducts,
   getSortedPriceProducts,
 } from "../../utilities";
+import axios from "axios";
 
 export function Product() {
   const { products } = useProducts();
   const { state } = useProductsFilter();
   const { sortByPrice, categories, price, rating } = state;
+  const { setCartItems } = useCart();
+
+  const addToCartHandler = async (product) => {
```

Comment:
again add into a seprate file as well as outside of the component

Reply:
this also will do in refactoring part